### DISCUSSION TITLE
Does `poetry update` not update git dependencies?

### DISCUSSION THREAD

**Discussion by truh** [python-poetry/poetry#9543] [Q&A] [answered]
In one of my projects I have a few dependencies as git dependencies. `package-name = { git = "git@github.com/owner/project.git" }`
I thought I could update them to latest commit on the main branch by running `poetry update package-name`. 20 minutes later and the `poetry.lock` file is unchanged.

I can edit the `resolved_reference` fields in the `poetry.lock` file to the latest revision and install with `poetry install` just fine.

**Comment by radoering** [comment, marked as answer]
Yeah, updating git dependencies does not work as far as I know. In the rare cases, I want to update a git dependency, I remove the complete `[package.source]` section of this dependency from the lock file before locking. Of course, PRs improving the update behavior of git dependencies are welcome.
